src/parser.ts

```diff
@@ -13,6 +13,21 @@ let PlaceHolders = {
     ArrowParameterPlaceHolder: 'ArrowParameterPlaceHolder'
 };
 
+function upgradeToken(token) {
+    if (token && state.strict && token.type === Token.Identifier) {
+        if (scanner.isStrictModeReservedWord(token.value)) {
+            token.type = Token.Keyword;
+        }
+    }
+    return token;
+}
+
+function nextToken() {
+    const token = scanner.lex();
+    scanner.lookahead = upgradeToken(scanner.lookahead);
```

Comment:
And thus why the goal of the next refactoring is to let the parser works on its own w.r.t strict mode reserved word with the need to taint the scanner at all.